src/yuzu/configuration/configuration_shared.h

```diff
@@ -15,8 +15,17 @@ constexpr int USE_GLOBAL_INDEX = 0;
 constexpr int USE_GLOBAL_SEPARATOR_INDEX = 1;
 constexpr int USE_GLOBAL_OFFSET = 2;
 
+enum CheckState {
```

Comment:
```suggestion
enum class CheckState {
```

This makes it strongly typed and prevents generic names from entering the global namespace. It also allows the enum to be forward declared.